depmap-compute/depmap_compute/context.py

```diff
@@ -28,14 +31,128 @@
 
 
 class ContextEvaluator:
+    """
+    Instantiated for a specific context. 
+    Caches data for the slices referenced in the context (in memory).
+    For context examples, see tests.
+    """
+
+    def __init__(
+        self, context: dict, get_slice_data: Callable[[SliceQuery], pd.Series],
+    ):
+        """
+        A `context` dict should have:
+            - a `dimension_type` such as "depmap_model" (which is not used in this evaluator)
+            - an `expr` such as { "==": [ { "var": "slice/lineage/1/label" }, "Breast" ] }
+        """
+        self.expr = _encode_dots_in_vars(context["expr"])
+        self.slice_query_vars = context.get("vars", {})
+
+        # Takes a slice query, returns a dictionary of slice values (indexed by ID)
+        self.get_slice_data = get_slice_data
+
+        # The cache is used so that slice values only need to be looked up once per context.
+        # - The keys in this dictionary are slice queries encoded as a tuple
+        #   (ex. `("Chronos_combined", "SOX10", "feature_label")`).
+        # - The values are each an entire dictionary of slice values (indexed by given ID)
+        self.cache = {}
+
+    def is_match(self, given_id: str):
```

Comment:
It doesn't have to be part of this PR. I liked the idea you had of passing a list of IDs and getting back a filtered list. Something like:
```
  def evaluate(self, given_ids: List[str]): List[str]
```
Or at that point this really wouldn't have to be class at all and could just be an `evaluate_context()` function.